python/dask_cudf/dask_cudf/io/csv.py

```diff
@@ -130,11 +133,14 @@ def _internal_read_csv(path, chunksize="256 MiB", **kwargs):
                     "names"
                 ] = meta.columns  # no header in the middle of the file
                 kwargs2["header"] = None
+            kwargs2["usecols"] = usecols
             dsk[(name, i)] = (apply, _read_csv, [fn, dtypes], kwargs2)
 
             i += 1
 
     divisions = [None] * (len(dsk) + 1)
+    if usecols is not None:
+        meta = meta[usecols]
```

Comment:
Just for my understanding - why are we able to pass `usecols` to `_read_csv` above when it is null but not `meta`?

Reply:
We wan't to fetch all the columns from the csv file. Hence skipped the passing of `usecols` in `meta` above.

Reply:
Cool! Thanks for the explanation 🙂